### PR TITLE
allow setting pool name

### PR DESCRIPTION
**Pull request by BusyJay**
Currently, all spawned threads in pool share father's thread name. It may be more friendly if we can distinguish between threads in pool and father's thread by setting a thread pool name.

### CODE REVIEW

**Comment by frewsxcv** [comment]
Thanks for the pull request! I'm not sure I see the point to adding an associated `String` for each `ThreadPool`. If all `Threads` within a `ThreadPool` share the same name, I'm not sure I see a point to having it personally. I'm aiming for this library to be fairly minimal, only include features directly related to the `ThreadPool` and want to avoid feature creep. What do you think?

**Comment by siddontang** [comment]
Sometimes, we need to know which thread takes more resource like CPU/MEM with `top -p ID -H`, but it is hard to check now because the thread in the pool has the same name with main thread. 

Btw, we may use one thread pool for some particular tasks, and use another thread pool for others, and we can distinguish these threads with different names.

Of course, if rust supports changing thread name dynamically, this PR is no needed.

**Comment by frewsxcv** [comment]
Will look at this again sometime later today or tomorrow (pretty busy today). Thanks for being patient!

**Comment by siddontang** [comment]
Or maybe we can use a default name to be separated with main thread first, I think it is ok too. :-)

File: src/lib.rs [hunk 1]
```diff
@@ -102,6 +110,14 @@ impl ThreadPool {
     ///
     /// This function will panic if `threads` is 0.
     pub fn new(threads: usize) -> ThreadPool {
+        ThreadPool::new_pool(None, threads)
+    }
+
+    pub fn new_with_name(name: String, threads: usize) -> ThreadPool {
```
Comment (frewsxcv) [inline src/lib.rs]: Can we call this just `with_name`?

Comment (frewsxcv) [inline src/lib.rs]: Actually, I'm fine with the name. Up to you to change it.

**Comment by frewsxcv** [comment]
I'll rebase this PR.

**Comment by frewsxcv** [comment]
Rebased and cleaned up your changes: https://github.com/frewsxcv/rust-threadpool/pull/28

**Pull request closed without merging**